SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -10,65 +10,111 @@
+	RemoveEditorAssetLoadedCallback();
 
-	PreProcessSchemaMap();
+	TArray<TStrongObjectPtr<UObject>> LoadedAssets;
+	if (bFullScan)
+	{
+		Progress.EnterProgressFrame(80.f);
+		if (!LoadPotentialAssets(LoadedAssets))
+		{
+			bSchemaGeneratorRunning = false;
+			LoadedAssets.Empty();
+			CollectGarbage(RF_NoFlags, true);
+			return false;
+		}
+	}
 
 	// Compile all dirty blueprints
 	TArray<UBlueprint*> ErroredBlueprints;
 	bool bPromptForCompilation = false;
 	UEditorEngine::ResolveDirtyBlueprints(bPromptForCompilation, ErroredBlueprints);
 
-	LoadDefaultGameModes();
+	TryLoadExistingSchemaDatabase();
 
-	SchemaGeneratorResult = Async<bool>(EAsyncExecution::Thread, SpatialGDKGenerateSchema,
-		[this, bCachedSpatialNetworking, ErroredBlueprints, SuccessCallback, FailureCallback]()
+	Progress.EnterProgressFrame(bFullScan ? 10.f : 100.f);
+	bool bResult = SpatialGDKGenerateSchema();
+
+	if (bFullScan)
+	{
+		Progress.EnterProgressFrame(10.f);
+		LoadedAssets.Empty();
+		CollectGarbage(RF_NoFlags, true);
+	}
+	
+	// We delay printing this error until after the schema spam to make it have a higher chance of being noticed.
+	if (ErroredBlueprints.Num() > 0)
 	{
-		// We delay printing this error until after the schema spam to make it have a higher chance of being noticed.
-		if (ErroredBlueprints.Num() > 0)
+		UE_LOG(LogSpatialGDKEditor, Error, TEXT("Errors compiling blueprints during schema generation! The following blueprints did not have schema generated for them:"));
+		for (const auto& Blueprint : ErroredBlueprints)
 		{
-			UE_LOG(LogSpatialGDKEditor, Error, TEXT("Errors compiling blueprints during schema generation! The following blueprints did not have schema generated for them:"));
-			for (const auto& Blueprint : ErroredBlueprints)
-			{
-				UE_LOG(LogSpatialGDKEditor, Error, TEXT("%s"), *GetPathNameSafe(Blueprint));
-			}
+			UE_LOG(LogSpatialGDKEditor, Error, TEXT("%s"), *GetPathNameSafe(Blueprint));
 		}
+	}
 
-		if (!SchemaGeneratorResult.IsReady() || SchemaGeneratorResult.Get() != true)
-		{
-			FailureCallback.ExecuteIfBound();
+	GetMutableDefault<UGeneralProjectSettings>()->bSpatialNetworking = bCachedSpatialNetworking;
+	bSchemaGeneratorRunning = false;
+
+	return bResult;
+}
+
+bool FSpatialGDKEditor::LoadPotentialAssets(TArray<TStrongObjectPtr<UObject>>& OutAssets)
+{
+	FAssetRegistryModule& AssetRegistryModule = FModuleManager::LoadModuleChecked<FAssetRegistryModule>("AssetRegistry");
+
+	TArray<FAssetData> FoundAssets;
+	AssetRegistryModule.Get().GetAllAssets(FoundAssets, true);
+
+	// Filter assets to game blueprint classes that are not loaded.
+	FoundAssets = FoundAssets.FilterByPredicate([](FAssetData Data) {
+		return (!Data.IsAssetLoaded() && Data.TagsAndValues.Contains("GeneratedClass") && Data.PackagePath.ToString().StartsWith("/Game"));
+	});
+
+	FScopedSlowTask Progress((float)FoundAssets.Num(), FText::FromString(FString::Printf(TEXT("Loading %d Assets before generating schema"), FoundAssets.Num())));
```

Comment:
Do we have a standard on (C-style) vs `static_cast`?

Reply:
static_cast